src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -92,15 +92,28 @@ private BoundExpression RewriteStringConcatenation(SyntaxNode syntax, BinaryOper
                     break;
 
                 case 2:
-                    result = RewriteStringConcatenationTwoExprs(syntax, leftFlattened.ToImmutable());
+                    var left = leftFlattened[0];
+                    var right = leftFlattened[1];
+                    result = RewriteStringConcatenationTwoExprs(syntax, left, right);
                     break;
 
                 case 3:
-                    result = RewriteStringConcatenationThreeExprs(syntax, leftFlattened.ToImmutable());
+                    {
+                        var first = leftFlattened[0];
```

Comment:
I feel either approach is OK. It's just a matter of calling ImmutableArray.Create a little bit earlier or a little bit later.

I don't find introducing a new scope below the case label particularly concerning, but you can get around it by just passing the `leftFlattened[n]` exprs to the call directly.

Reply:
Cool, I'll leave as-is.